guardian/master-to-main
# Convert to using the rename a branch API

## What does this change?

This PR moves from defining the required code to run the migration process to calling the [rename a branch](https://docs.github.com/en/rest/reference/repos#rename-a-branch) API that is now available. The reduces the amount of code required substantially and also removes the burden of maintaining the correct migration process from this repository. The tool still provides some checks and some post-migration tasks so may still be useful for migration Guardian specific repositories.

## How to test

Use the new version of the tool to migrate the https://github.com/guardian/master-to-main-demo repo and see that it works as expected.

## How can we measure success?

The migration process works as expected with much less code for us to maintain.

## Have we considered potential risks?

Changes could introduce new bugs which leave repositories in broken state. As we're using the GitHub API to do the vast majority of the work now, this is far less likely.
